package/MDAnalysis/analysis/bat.py

```diff
@@ -0,0 +1,422 @@
+        ValueError
+            If ag contains more than one molecule
+
+        """
+        super(BAT, self).__init__(ag.universe.trajectory, **kwargs)
+        self._ag = ag
+
+        # Check that the ag contains bonds
+        if not hasattr(ag, 'bonds'):
+            raise AttributeError('AtomGroup has no attribute bonds')
+
+        # Determine the root
+        # The initial atom must be a terminal atom
+        terminal_atoms = _sort_atoms_by_mass(\
+          [a for a in self._ag.atoms if len(a.bonds)==1])
+        if (initial_atom is None):
+            # Select the heaviest root atoms from the heaviest terminal atom
+            initial_atom = terminal_atoms[-1]
+        elif (not initial_atom in terminal_atoms):
+            raise Exception('Initial atom is not a terminal atom')
+        # The next atom in the root is bonded to the initial atom
+        second_atom = initial_atom.bonded_atoms[0]
+        # The last atom in the root is the heaviest atom
+        # bonded to the second atom
+        third_atom = [a for a in second_atom.bonded_atoms \
+          if (a in self._ag) and (a!=initial_atom)][-1]
+        self._root = mda.AtomGroup([initial_atom, second_atom, third_atom])
+
+        # Construct a list of torsion angles
+        self._torsions = []
+        selected_atoms = mda.AtomGroup(self._root)
+        while len(selected_atoms) < self._ag.n_atoms:
+            try:
+                torsion = _find_torsion(selected_atoms, self._ag)
+            except:
+                raise ValueError('AtomGroup is more than one molecule')
+
+            self._torsions.append(torsion)
+            selected_atoms += torsion[0]
+
+        # Get indices of the root and torsion atoms
+        # in a Cartesian positions array that matches the AtomGroup
+        ag_top_inds = list(self._ag.indices)
+        self._root_XYZ_inds = [ag_top_inds.index(a.index) for a in self._root]
+        self._torsion_XYZ_inds = [[ag_top_inds.index(a.index) for a in t] \
+          for t in self._torsions]
+
+        # The primary torsion is the first torsion on the list
+        # with the same central atoms
+        prior_atoms = [sorted([a1, a2]) for (a0, a1, a2, a3) in self._torsions]
+        self._primary_torsion_indices = [prior_atoms.index(prior_atoms[n]) \
+          for n in range(len(prior_atoms))]
+
+        self._ag1 = mda.AtomGroup([ag[0] for ag in self._torsions])
+        self._ag2 = mda.AtomGroup([ag[1] for ag in self._torsions])
+        self._ag3 = mda.AtomGroup([ag[2] for ag in self._torsions])
+        self._ag4 = mda.AtomGroup([ag[3] for ag in self._torsions])
+
+    def _prepare(self):
+        self.bat = []
+
+    def _single_frame(self):
+        # Calculate coordinates based on the root atoms
+        # The rotation axis is a normalized vector pointing from atom 0 to 1
+        # It is described in two degrees of freedom
+        # by the polar angle and azimuth
+        (p0, p1, p2) = self._root.positions
+        v01 = p1 - p0
```

Comment:
So this assumes that p0, p1, p2 are all in the same periodic image right?  If the p1-p2 bond was wrapped over a box boundary what would happen?

Reply:
Yes, that is an assumption. I will use MDAnalysis.lib._cutil.make_whole to unwrap the positions of the root atoms.